src/prommis/uky/uky_flowsheet.py

```diff
@@ -861,5 +873,956 @@ def display_results(m):
+            flowsheet.fs.roaster.flow_mol_comp_product[0, "Gd"]
+            * 362.5
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
+        ),
+        units=units.kg / units.hr,
+        mutable=True,
+        doc="Product gadolinium mass flow",
+    )
+
+    m.fs.La_product = Param(
+        default=units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_product[0, "La"]
+            * 325.82
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
+        ),
+        units=units.kg / units.hr,
+        mutable=True,
+        doc="Product lanthanum mass flow",
+    )
+
+    m.fs.Nd_product = Param(
+        default=units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_product[0, "Nd"]
+            * 336.48
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
+        ),
+        units=units.kg / units.hr,
+        mutable=True,
+        doc="Product neodymium mass flow",
+    )
+
+    m.fs.Pr_product = Param(
+        default=units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_product[0, "Pr"]
+            * 329.82
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
+        ),
+        units=units.kg / units.hr,
+        mutable=True,
+        doc="Product praseodymium mass flow",
+    )
+
+    m.fs.Sc_product = Param(
+        default=units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_product[0, "Sc"]
+            * 137.912
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
+        ),
+        units=units.kg / units.hr,
+        mutable=True,
+        doc="Product scandium mass flow",
+    )
+
+    m.fs.Sm_product = Param(
+        default=units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_product[0, "Sm"]
+            * 348.72
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
+        ),
+        units=units.kg / units.hr,
+        mutable=True,
+        doc="Product samarium mass flow",
+    )
+
+    m.fs.Y_product = Param(
+        default=units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_product[0, "Y"]
+            * 225.812
+            * units.g
+            / units.mol,
+            to_units=units.kg / units.hr,
+        ),
+        units=units.kg / units.hr,
+        mutable=True,
+        doc="Product yttrium mass flow",
+    )
+
+    pure_product_output_rates = {
+        "CeO2": m.fs.Ce_product,
+        "Sc2O3": m.fs.Sc_product,
+        "Y2O3": m.fs.Y_product,
+        "La2O3": m.fs.La_product,
+        "Nd2O3": m.fs.Nd_product,
+        "Pr6O11": m.fs.Pr_product,
+        "Sm2O3": m.fs.Sm_product,
+        "Gd2O3": m.fs.Gd_product,
+        "Dy2O3": m.fs.Dy_product,
+    }
+
+    m.fs.Ce_dust = Param(
+        default=units.convert(
+            flowsheet.fs.roaster.flow_mol_comp_dust[0, "Ce"]
```

Comment:
Just want to verify that the "dust" streams are supposed to be product streams and not volatile/dust waste. The UKy report define pure products as targeted oxides that should be separated into pure streams, and mixed products as other oxides that have notable value but may be solid as a mixed basket or are not worth separating.

Reply:
So are you saying the dust streams should be mixed products (the current implementation) or pure product?

Reply:
The roaster produces a waste dust and also a sintered product. It seems from https://github.com/prommis/prommis/blob/main/src/prommis/roasting/ree_oxalate_roaster.py#L41 that `flow_mol_comp_dust` is a waste stream and part of the dust & volatiles waste flow. The solid product contains both pure and mixed products, which are separated downstream into pure REO streams and a mixed basket of everything else. I don't think we have the downstream units here yet, so you could consider assigning a few specific components as "pure"/"desired" products and everything else as "mixed" for now.

Reply:
The roaster does not really produce both mixer and pure products - it just produces a mixer product which can be further separated later (it is not like there are two separate phases of material). For the UKy process, the only product is mixer REEs (which is the desired product); there is no "pure" product of interest here.

Separation of the mixer REEs into more targeted grouping required an entire additional process that we are not modeling here (and would often happen at a different geographic location).

Reply:
@andrewlee94 thank you for the explanation. @MarcusHolly you can assign all products as mixed products then. Set the argument `pure_product_output_rates` as an empty dictionary, or a single component with zero flowrate. I made the pure and mixed product rates required arguments and an error is thrown if one of them is not set so that users know this is something that should be considered.

Reply:
I just set it as an empty dictionary.